services/director-v2/src/simcore_service_director_v2/models/schemas/dynamic_services/scheduler.py

```diff
@@ -375,6 +376,14 @@ def from_service_inspect(
         labels = service_inspect["Spec"]["Labels"]
         return cls.parse_raw(labels[DYNAMIC_SIDECAR_SCHEDULER_DATA_LABEL])
 
+    def as_label_data(self) -> str:
+        # compose_spec needs to be json encoded
+        # before encoding it to json and storing it
+        # in the label
+        scheduler_data_copy = deepcopy(self)
+        scheduler_data_copy.compose_spec = json.dumps(scheduler_data_copy.compose_spec)
+        return scheduler_data_copy.json()
+
```

Comment:
Yes, ``Json`` annotation **loads as json string** BUT **exports as dict** ... 

I understand what you intend to do but I wonder if there is a more elegant way to proceed :-) since by reassigning ``scheduler_data_copy.compose_spec`` you are avoiding the parsing mechanisms (you take advantage that by default the validation is only upon construction)

TIP: this can all be done efficiently in a single line taking advantage of the pydantic API
```python
return self.copy(deep=True, update={"compose_spec": self.json(include={"compose_spec"}) })

```

Reply:
Not sure about `efficiently` since you go through pedantic's validation and internals as to respect to deepcopy.

Below correct solution. I've added a regression test.

```python
return self.copy(
            update={"compose_spec": json.dumps(self.compose_spec)}, deep=True
        ).json()
```

Reply:
There is no validation above (notice that you are not constructing anything and decorator ``always=False``) and in addition pydantic is cythonized. In any case, if you time it we can be sure what is more efficient.

Reply:
really? :-)  I thought ``json.dumps(self.compose_spec)`` would be exactly the same as ``self.json(include={"compose_spec"})``

Reply:
I would loved for this to work! Unfortunately  `compose_spec` is of type `Optional[Dict[str, Any]]`. Will have to keep the current version.